webui/react/src/components/WorkspaceCreateModal.tsx

```diff
@@ -106,23 +83,20 @@ const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }
         );
         return clusterNamespacePairs.namespaceBindings;
       } catch (e) {
-        if (!isNonK8RMError(e)) {
-          handleError(e, {
-            level: ErrorLevel.Error,
-            publicMessage: 'Failed to fetch list of workspace namespace bindings.',
-            silent: false,
-            type: ErrorType.Server,
-          });
-        }
+        handleError(e, {
+          level: ErrorLevel.Error,
+          publicMessage: 'Failed to fetch list of workspace namespace bindings.',
+          silent: true,
+          type: ErrorType.Server,
+        });
         return NotLoaded;
       }
     },
-    [workspaceId],
+    [resourceManagers, workspaceId],
   );
-
   const resourceQuotasList = useAsync(
     async (canceller) => {
-      if (workspaceId === undefined) {
+      if (workspaceId === undefined || Loadable.getOrElse([], resourceManagers).length <= 0) {
```

Comment:
nit
```suggestion
      if (workspaceId === undefined || Loadable.getOrElse([], resourceManagers).length == 0) {
```
---

questions: i see `workspaceId === undefined || Loadable.getOrElse([], resourceManagers).length <= 0` prevent APIs in `useAsync` to be called many times. Just wondering if its safe enough. are there any situations that pass through the conditions and end up calling many API calls?

Reply:
this is a good point -- i do think we need to keep whether or not the modal is open as a prop because once `resourceManagers` returns a non-zero length, all the other modals in the sidebars will begin to make requests.

Reply:
basically my suggestion was to use the `workspaceId` check to function the same as the `open` prop check -- it's only defined when the user opens an edit modal, then goes back to `undefined` `onClose`. always `undefined` on create modals.

Reply:
nit: in addition to changing length check, we could consolidate all of these to one reused var, something like `const resourceManagers = Loadable.getOrElse([], loadableResourceManagers).length === 0`